### ISSUE TITLE
E0110 False positive when using attrs.field

### ISSUE THREAD

**Issue by dbramel** [pylint-dev/pylint#9013]
### Bug description

When defining an abstract class with an abstract property and using `attrs` to define a concrete type, pylint will report a false-positive E0110 on instantiating the concrete type if the property is initialized with `attrs.field()`:

```python
from __future__ import annotations
from abc import ABC, abstractmethod
import attrs


class Abstract(ABC):
    @property
    @abstractmethod
    def my_prop(self) -> int:
        ...


@attrs.define
class Concrete(Abstract):
    my_prop: int


@attrs.define
class Concrete_FalsePositive(Abstract):
    my_prop: int = attrs.field()


conc = Concrete(35)
conc_fp = Concrete_FalsePositive(99)

print(f"conc:{conc.my_prop} fp:{conc_fp.my_prop}")
```

### Command used

```shell
pylint ./test_pylint_bug.py
```

### Pylint output

```shell
************* Module test_pylint_bug
test_pylint_bug.py:25:10: E0110: Abstract class 'Concrete_FalsePositive' with abstract methods instantiated (abstract-class-instantiated)

------------------------------------------------------------------
Your code has been rated at 6.15/10 (previous run: 6.15/10, +0.00)
```

### Expected behavior

No error should be reported

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.10.13 (main, Aug 28 2023, 08:10:03) [GCC 11.2.0]
```

### OS / Environment

GNU/Linux
